File: TicketSystem/Ticket.java

public class Ticket implements Comparable<Ticket> {

    private int priority;
    private int seat;
    private int row;
    private String stand;
    private String name;
    private double price;

    public Ticket(String name, double price, int seat, int row, int priority){

        this.name = name;
        this.price = price;
        this.seat = seat;
        this.row = row;
        this.priority = priority;
        //this.stand = stand;

    }

    public int getPriority(){
        return priority;
    }

    public int getSeat(){
        return seat;

    }

    public String getStand(){
        return stand;
    }

    public String getName(){

        return name;
    }
    public double getPrice(){
        return price;
    }

    public void setPriority(int priority){

        this.priority = priority;
    }

    @Override
    public String toString(){

        return(name + " " + "$" + price + " Seat: " + seat + " Row: " +  row + " Priority: " + priority);

    }

    @Override

    public int compareTo(Ticket other) {
        return Integer.compare(this.priority, other.priority);
    }
    public String fileString(){

        return(name + " " + price + " " + row + " " + seat + " " + priority);

    }

}

File: Learning/TicketSystem/Ticket.cpp
//
// Created by Hyacinthe Chemasle on 15/12/2024.
//

#include "Ticket.h"
#include <sstream>
#include <iostream>
#include <string>

Ticket::Ticket(std::string name, double price, int seatNumber, int rowNumber) //Constructor for Ticket Class
 : name(name), price(price), seatNumber(seatNumber), rowNumber(rowNumber){} //this is specific C++ syntax

/*These are the methods which are all Getters. The syntax is almost the same to Java*/

std::string Ticket::getName(){
  return name;
}

double Ticket::getPrice(){
  return price;
}

int Ticket::getSeatNumber(){
  return seatNumber;
}

int Ticket::getRowNumber(){
  return rowNumber;
}

/*toString method equivalent in C++*/

std::string Ticket::toString(){
  std::ostringstream sb; //Creates StringBuilder oss = new StringBuilder();
  sb << name << "$" << price << " Seat: " << seatNumber << " Row: " << rowNumber; // << equivalent to + but in the context of StringBuilder it refers to oss.append(name);
  return sb.str(); //equivalent to oss.toString(); sb.toString();
}


